ci/ray_ci/doc/module.py

```diff
@@ -0,0 +1,63 @@
+import importlib
+import inspect
+from types import ModuleType
+from typing import Set
+
+
+class Module:
+    def __init__(self, module: str):
+        self._module = importlib.import_module(module)
+        self._visited = set()
+        self._class_apis = set()
+        self._function_apis = set()
+
+    def walk(self) -> None:
+        self._walk(self._module)
+
+    def get_class_apis(self) -> Set:
+        self.walk()
+        return self._class_apis
+
+    def get_function_apis(self) -> Set:
+        self.walk()
+        return self._function_apis
+
+    def _walk(self, module: ModuleType) -> None:
+        """
+        Depth-first search through the module and its children to find annotated classes
+        and functions.
+        """
+        if module in self._visited:
+            return
+        self._visited.add(module)
```

Comment:
should this `_visited` be saving a path/name or something, rather than the reference? can this be forcing the parser to save all `ModuleType` in the end and preventing it from garbage collected, which might be heavier than we think?